packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -28,13 +28,7 @@ export type InsightsProps<
   TInsightsClient extends ProvidedInsightsClient = ProvidedInsightsClient
 > = {
   insightsClient?: TInsightsClient;
-  insightsInitParams?: {
-    userHasOptedOut?: boolean;
-    useCookie?: boolean;
-    anonymousUserToken?: boolean;
-    cookieDuration?: number;
-    region?: 'de' | 'us';
-  };
+  insightsInitParams?: Omit<InsightsMethodMap['init'][0], 'appId' | 'apiKey'>;
```

Comment:
```suggestion
  insightsInitParams?: Partial<InsightsMethodMap['init'][0]>;
```

it seems to allow override of all params in theory